### DISCUSSION TITLE
ROS2 Topics Disappear After Connecting RPi to Network via Ethernet

### DISCUSSION THREAD

**Discussion by briellelaw** [iRobotEducation/create3_docs#439] [Questions]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi 4, Ubuntu 22.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.2

### Which RMW is your robot running?

CycloneDDS

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

My current setup:
- iRobot Create3 connected to (and powering) an RPi4 via Power2Go cable
- rplidar connected to RPi4 via USB
- RPi4 connected to network via Ethernet to Ethernet Wall Port 
- I've tried both fast_dds and cyclone_dds as the middleware

Problem 1:
I run:
 `ros2 topic list --no-daemon --time-spin 10` 

I can only see all the ros topics when there is No network connection via ethernet. 
```
/battery_state
/cliff_intensity
/cmd_audio
/cmd_lightring
/cmd_vel
/clock_status
/hazard_decection
/imu
/interface_buttons
/ir_intensity
/ir_opcode
/kidnap_status
/mobility_monitor/transition_event
/mouse
/odom
/parameter_events
/robot_state/transition_event
/rosout
/scan
/slip_status
/static_transform/transition_event
/stop_status
/tf
/tf_static
/wheel_status
/wheel_ticks
/wheel_vels
```

Once I connect the Ethernet cable, I only see the following topics:
```
/parameter_events 
/rosout
```

**Comment by kwujciak** [comment]
Is your Pi/robot also connected to internet? My guess is when it's disconnected, your Pi is seeing the topics over internet. I've had an experience where if I connected my Pi via usb ethernet, sometimes it assigns a different usb number other than 0. If it read "usb device 0" in the logs, I was able to pull a full topic list over ethernet. If it read "usb device 3" (or some other number), I would not be able to see the topic list, but rebooting the robot helped.
